contracts/NodeDelegator.sol

```diff
@@ -338,6 +393,169 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
+    /// Transfers the withdrawn assets to the staker that requested the withdrawal.
+    /// Is only callable by the `LRTDepositPool` contract.
+    /// @param withdrawal the `withdrawal` data emitted in the `WithdrawalQueued` event from EigenLayer's
+    /// `DelegationManager` contract when `requestWithdrawal` was called on the `LRTDepositPool` contract by the staker.
+    /// @param staker the address of the staker requesting the withdrawal
+    /// @return asset address of the liquid staking tokens (LST) that were claimed.
+    /// @return assets the amount of LSTs received from the withdrawal
+    function claimWithdrawal(
+        IDelegationManager.Withdrawal calldata withdrawal,
+        address staker
+    )
+        external
+        onlyDepositPool
+        whenNotPaused
+        returns (address asset, uint256 assets)
+    {
+        // Make sure the staker requested this withdrawal
+        // and the withdrawal was not manipulated
+        bytes32 withdrawalRoot = _calculateWithdrawalRoot(withdrawal);
+        if (withdrawalRequests[withdrawalRoot] != staker) {
+            revert StakersWithdrawalNotFound();
+        }
+
+        IERC20[] memory tokens = new IERC20[](1);
+        tokens[0] = IStrategy(withdrawal.strategies[0]).underlyingToken();
+        asset = address(tokens[0]);
+
+        // There can be assets sitting in this NodeDelegator contract so we need to account for those.
+        uint256 assetsBefore = IERC20(asset).balanceOf(address(this));
+
+        // Claim the previously requested withdrawal from EigenLayer
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
+        // the 3rd middlewareTimesIndexes param is not used in the EigenLayer M2 contracts
+        IDelegationManager(delegationManagerAddress).completeQueuedWithdrawal(withdrawal, tokens, 0, true);
+
+        // Calculate the amount of assets returned from the withdrawal
+        assets = IERC20(asset).balanceOf(address(this)) - assetsBefore;
+        if (assets > 0) {
+            // transfer withdrawn assets to the staker
+            IERC20(asset).transfer(staker, assets);
+        }
+    }
+
+    /// @notice Claims the previously requested internal withdrawal from the underlying EigenLayer strategy.
+    /// Transfers the withdrawn liquid staking tokens to the `LRTDepositPool` contract.
+    /// Is only callable by accounts with the Operator role.
+    /// @param withdrawal the `withdrawal` data emitted in the `WithdrawalQueued` event from EigenLayer's
+    /// `DelegationManager` contract when `requestInternalWithdrawal` was called
+    /// on the `NodeDelegator` contract by a Prime Operator.
+    /// @return asset address of the liquid staking tokens (LST) that were claimed.
+    /// @return assets the amount of LSTs transferred to the `LRTDepositPool` contract.
+    function claimInternalWithdrawal(IDelegationManager.Withdrawal calldata withdrawal)
```

Comment:
Only code change I see needed from the full code review:

None of the internal withdraw functions (undelegate, requestInternalWithdrawal, claimInternalWithdrawal) emit events. These actions would be useful to be able to report on from monitoring.

Reply:
Here's the events I added
```Solidity
event RequestWithdrawal(
    address indexed strategy, bytes32 indexed withdrawalRoot, address indexed staker, uint256 strategyShares
);
event ClaimWithdrawal(
    address indexed strategy, bytes32 indexed withdrawalRoot, address indexed staker, uint256 assets
);
event Delegate(address indexed operator);
event Undelegate(address indexed strategy, uint256 strategyShares);
```